README.md

```diff
@@ -59,22 +72,12 @@ Note, when using app links you do not need a web server anymore that redirects t
 All you need to host is a `manifest.json` file for the app details and the assetlinks.json file for the app links.
 
 ### Thread Handling
-The j2v8 engine requires that calls to the Blockstack session are made from only one thread.
-
-The SDK comes with a default implementation of an `Executor` that manages thread switching. 
-By default, network threads are done in the background, calls to the Blockstack session are done
-on the main thread.
-
-If the Blockstack session is not created on the main thread then a custom implementation of `Excecutor`
-needs to be provided in the constructor of the Blockstack session. See the [service example](/example-service) for some code.   
-
-It is also possible to manually switch threads buy using `.releaseThreadLock` and `.aquireThreadLock`.
-These methods allow to make calls to the Blockstack session on a different thread. The thread lock
-needs to be released on the current thread of the session. Then the new thread can acquire the thread lock.
+The Android SDK uses kotlin's Coroutines. Network requests are using the `Dispatchers.IO` context.
 
 ### Sign-In Flow
-The most basic way to sign-in a user with Blockstack is to use `redirectUserToSignIn`, 
-`handlePendingAuthResponse` and all subsequent method calls in the same activity. This is shown in the 
+The most basic way to sign-in a user with Blockstack is to use `BlockstackSignIn.redirectUserToSignIn`, 
```

Comment:
This is confusing.

```
The most basic way to sign-in a user with Blockstack is to use `BlockstackSignIn.redirectUserToSignIn`, 
`handlePendingAuthResponse`, and all subsequent method calls in the same activity. This is shown in the 	`BlockstackSession.handlePendingAuthResponse` and all subsequent method calls in the same activity. 
This is shown in the 
[simple example](/example). However, applications usually have a separate screen to handle user sessions. 
```	

`and all subsequent method calls in the same activity` doesn't seem to refer to the first half of the sentence. Can you clarify what you mean?

Also, "This is shown" --- what does "this" refer to --- the activity? the sign-in? Something else?